Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAutomationWorker.py

```diff
@@ -18,28 +25,114 @@ def init_locals(WorkspaceId, RegDomain):
 def Set_Marshall(WorkspaceId, Enabled, RegDomain):
     (WorkspaceId, RegDomain) = init_locals(WorkspaceId, RegDomain)
     if (Enabled):
-        i = 1
+        #call the registration script
+        try:
+            agent_id = read_oms_config_file()
+            sp = subprocess.call(
+                ["python", REGISTRATION_FILE_PATH, "--register", "-w " + WorkspaceId, "-a " + agent_id,
+                 "-c " + OMS_CERTIFICATE_PATH, "-k " + OMS_CERT_KEY_PATH, "-p " + WORKING_DIRECTORY_PATH])
+            if sp != 0:
+                LG().Log('ERROR', "Linux Hybrid Worker registration failed")
+                return [-1]
+            if not os.path.isfile(WORKER_CONF_PATH):
+                LG().Log('ERROR', "Linux Hybrid Worker registration file could not be created")
+                return [-1]
+        except Exception as exception:
+            LG().Log('ERROR', exception.message)
+
+        # Read the worker state file and try to kill linux hybrid worker process if running
+        try:
+            (pid, workspace_id) = read_worker_state()
+        except Exception as exception:
+            LG().Log('ERROR', exception.message)
+        try:
+            command = subprocess.check_output(["ps", "-p", str(pid), "-o", "comm="])
+        except subprocess.CalledProcessError:
+            pass
```

Comment:
The process with that PID may or may not be running, if it isn't running, then there is nothing to kill or do.